ddtrace/tracer/spancontext.go

```diff
@@ -406,47 +414,49 @@ func (t *trace) finishedOne(s *span) {
 		// TODO(barbayar): make sure this doesn't happen in vain when switching to
 		// the new wire format. We won't need to set the tags on the first span
 		// in the chunk there.
-		t.setTraceTags(s)
-	}
-	tr, ok := internal.GetGlobalTracer().(*tracer)
-	if !ok {
-		return
+		t.setTraceTags(s, tr)
 	}
-	// we have a tracer that can receive completed traces.
-	atomic.AddUint32(&tr.spansFinished, uint32(len(t.spans)))
-	if tr.config.partialFlushMinSpans > 0 && t.finished >= tr.config.partialFlushMinSpans && len(t.spans) != t.finished {
-		log.Debug("Partial flush triggered with %d finished spans", t.finished)
-		//TODO: is there a metric we should bump when doing this?
-		finishedSpans := make([]*span, 0, t.finished)
-		leftoverSpans := make([]*span, 0, len(t.spans)-t.finished)
-		for _, s2 := range t.spans {
-			if s2.finished {
-				s2.setMetric(keySamplingPriority, *t.priority) //TODO: maybe we don't have to do this for every span
-				finishedSpans = append(finishedSpans, s2)
-			} else {
-				leftoverSpans = append(leftoverSpans, s2)
-			}
-		}
-		t.spans = leftoverSpans
-		t.finished = 0
-		tr.pushChunk(&chunk{
-			spans:    finishedSpans,
+
+	if len(t.spans) == t.finished { // perform a full flush of all spans
+		t.finishChunk(tr, &chunk{
+			spans:    t.spans,
 			willSend: decisionKeep == samplingDecision(atomic.LoadUint32((*uint32)(&t.samplingDecision))),
 		})
-	}
-	if len(t.spans) != t.finished {
+		t.spans = nil
```

Comment:
Should we also be resetting t.finished here to zero?

Reply:
That happens in `t.finishChunk` since it's consistent between both implementations. LMK if you think it should just be moved out though.

Reply:
Oh good point, maybe we can also pull the `t.spans = ` line through as well? Pass in a `leftoverSpans` slice to `finishChunk` and in this case we pass nil since there are none leftover? I'm not sure how I feel about that though, feel free to say that's more confusing and just close this comment

Reply:
I think that might be slightly more confusing, but don't feel that strongly. For now, I'll resolve this comment, but we can explore changing this in the future if we find this is too brittle